tests/DoctrineObjectTest.php

```diff
@@ -2978,4 +2979,49 @@ public function testWrongEnumBackedValueThrowsException(): void
         $this->hydratorByValue->addStrategy('enum', new Assets\SimpleEnumStrategyPhp81());
         $this->hydratorByValue->hydrate($data, $entity);
     }
+
+    /**
+     * @requires PHP 8.1
+     */
+    public function testExtractReadonlyProps(): void
+    {
+        $entity = new Assets\SimpleEntityWithReadonlyPropsPhp81(2);
+
+        $this->configureObjectManagerForSimpleEntity(Assets\SimpleEntityWithReadonlyPropsPhp81::class);
+
+        $data = $this->hydratorByReference->extract($entity);
+        $this->assertEquals([], $data);
+
+        $entity->setField('value');
+        $data = $this->hydratorByReference->extract($entity);
+        $this->assertEquals(['field' => 'value'], $data);
+    }
+
+    /**
+     * @requires PHP 8.1
+     */
+    public function testHydrateReadonlyProps(): void
+    {
+        $this->configureObjectManagerForSimpleEntity(Assets\SimpleEntityWithReadonlyPropsPhp81::class);
+
+        $entity = $this->hydratorByValue->hydrate(['field' => 'toto', 'id' => 15], new Assets\SimpleEntityWithReadonlyPropsPhp81(3));
```

Comment:
```suggestion
        $entity = $this->hydratorByReference->hydrate(['field' => 'toto', 'id' => 15], new Assets\SimpleEntityWithReadonlyPropsPhp81(3));
```

This test should hydrate by reference. Hydrating by value, i.e. by calling getters/setters, will be fine, as long as the user doesn't add setters for a readonly property - which then would clearly be a user error. The case we should test here is that our library should not try to modify properties via reflection if they are readonly.

Reply:
done